server/controllers/AuthController.js

```diff
@@ -48,32 +49,35 @@ module.exports = {
             return;
         }
         let authorized = false;
-        if (req.session.authType === SIGN_UP) {
-            const userSaved = await services.UsersService.signupUser({
-                accessToken: req.session.accessToken,
-                user: userToAuthorize
-            });
-            if (userSaved) {
-                authorized = true;
-                userToAuthorize = userSaved;
-            }
-        } else if (req.session.authType === LOGIN) {
-            const { name: fullname, username, email } = userToAuthorize;
+        if (
+            req.session.authType === SIGN_UP ||
+            req.session.authType === LOGIN
+        ) {
+            let user;
 
-            const dbUser = await services.UsersService.getUserAndUpdateRoles({
-                accessToken: req.session.accessToken,
-                user: {
+            if (req.session.authType === SIGN_UP) {
+                user = await UsersService.signupUser({
```

Comment:
If you try to sign up a user that already exists, this function just returns that user. This is fine but confusing -- it seems like we should make it cleared in the name of the function.